src/downloader/chain_config.rs

```diff
@@ -0,0 +1,21 @@
+use anyhow;
+use serde::Deserialize;
+use std::collections::HashMap;
+use toml;
+
+pub struct ChainsConfig(pub HashMap<String, ChainConfig>);
+
+#[derive(Hash, Deserialize)]
+pub struct ChainConfig {
+    pub id: u32,
+    pub genesis: String,
```

Comment:
Please use specialized type: ethereum_types::H256

Reply:
It's OK and actually encouraged to use semantic types everywhere, including Config definitions. Embrace the type system! 🙂